mcfly/find_architecture.py

```diff
@@ -65,12 +68,17 @@ def train_models_on_samples(X_train, y_train, X_val, y_val, models,
     X_train_sub = X_train[:subset_size, :, :]
     y_train_sub = y_train[:subset_size, :]
 
+    metric_name = get_metric_name(metric)
+
     histories = []
-    val_accuracies = []
+    val_metrics = []
     val_losses = []
     for i, (model, params, model_types) in enumerate(models):
         if verbose:
             print('Training model %d' % i, model_types)
+        model_metrics = [get_metric_name(name) for name in model.metrics]
+        if metric_name not in model_metrics:
+            raise ValueError('Invalid metric, the model should be compiled with the same metric!')
```

Comment:
Error message is vague as it is. The message should include the currently selected metric at least. Also, I would prefer a period instead of an exclamation mark.